### PR TITLE
sambamba 2

### PR DESCRIPTION
**Pull request by northwestwitch**

### CODE REVIEW

File: lib/MIP/Program/Alignment/Sambamba.pm [hunk 1]
```diff
@@ -0,0 +1,697 @@
...
+    our @EXPORT_OK =
+      qw{sambamba_view sambamba_index sambamba_sort sambamba_markdup sambamba_flagstat sambamba_depth};
+}
+
+## MIPs lib/
+use lib catdir( dirname($Bin), 'lib' );
+use MIP::Unix::Standard_streams qw{unix_standard_streams};
+use MIP::Unix::Write_to_file qw{unix_write_to_file};
+
+## Constants
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Sambamba.pm]: Move to after line 36

File: lib/MIP/Program/Alignment/Sambamba.pm [hunk 2]
```diff
@@ -0,0 +1,697 @@
...
+## Arguments: $infile_path, $outfile_path, $stderrfile_path, $FILEHANDLE, temp_directory, $show_progress, $hash_table_size, $overflow_list_size, $io_buffer_size
+##          : $infile_path        => Infile path
+##          : $outfile_path       => Outfile path
+##          : $stderrfile_path    => Stderrfile path
+##          : $FILEHANDLE         => Sbatch filehandle to write to
+##          : $temp_directory     => Specify directory for temporary files
+##          : $show_progress      => Show progress
+##          : $hash_table_size    => Size of hash table for finding read pairs
+##          : $overflow_list_size => Size of the overflow list where reads, thrown from the hash table, get a second chance to meet their pairs
+##          : $io_buffer_size      => Two buffers of BUFFER_SIZE *megabytes* each are used for reading and writing BAM during the second pass
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Sambamba.pm]: Fix indent of fat comma

File: lib/MIP/Program/Alignment/Sambamba.pm [hunk 3]
```diff
@@ -0,0 +1,697 @@
...
+## Arguments: $depth_cutoffs_ref, $FILEHANDLE, $infile_path, $outfile_path, $stderrfile_path, $region, $filter, $min_base_quality, $mode, $fix_mate_overlap
+##          : $depth_cutoffs_ref => Multiple thresholds can be provided, for each one an extra column will be added, the percentage of bases in the region where coverage is more than this value {REF}
+##          : $FILEHANDLE        => Sbatch filehandle to write to
+##          : $infile_path       => Infile path
+##          : $outfile_path      => Outfile path
+##          : $stderrfile_path   => Stderrfile path
+##          : $region            => List or regions of interest or a single region in form chr:beg-end
+##          : $filter            => Set custom filter for alignments
+##          : $fix_mate_overlap  => Detect overlaps of mate reads and handle them on per-base basis
+##          : $min_base_quality  => don't count bases with lower base quality
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Sambamba.pm]: Captital letter in don't

File: lib/MIP/Program/Alignment/Sambamba.pm [hunk 4]
```diff
@@ -0,0 +1,697 @@
...
+
+    if ($mode) {
+
+        push @commands, $mode;
+    }
+    if ($region) {    # Limit output to regions
+
+        push @commands, q{--regions} . $SPACE . $region;
+    }
+    if (@$depth_cutoffs_ref) {
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Sambamba.pm]: @$depth_cutoffs_ref _> @{ $depth_cutoffs_ref }

File: t/sambamba_depth.t [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+## Specific arguments
+my %specific_argument = (
+
+    stderrfile_path => {
+        input           => q{stderrfile.test},
+        expected_output => q{2> stderrfile.test},
+    },
+    depth_cutoffs_ref => {
+        inputs_ref      => [q{25 100}],
```
Comment (henrikstranneheim) [inline t/sambamba_depth.t]: Should this not be qw{25 100}?

Comment (henrikstranneheim) [reply t/sambamba_depth.t]: or is it just a single element?

File: lib/MIP/Program/Alignment/Sambamba.pm [hunk 6]
```diff
@@ -0,0 +1,697 @@
+package MIP::Program::Alignment::Sambamba;
+
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Alignment/Sambamba.pm]: Excellent! Just some small changes. Could you also add the tests to t/config/run_tests.t (in alphabetical order)

**Comment by northwestwitch** [comment]
The problem with the following code is that if I put a string "25 100"
 then the code works, but I know that it's supposed to accept an array.

I think the issue might be in the code of Sambamba.pm itself:

as i is now in develop:

*push @commands, "--cov-threshold ".join(" --cov-threshold ", @{
$depth_cutoffs_ref }));*

if you give it an array it produces:  "*--cov-threshold 25 --cov-threshold
100*"


I think the code should be modified into:

*push @commands, q{--cov-threshold} . $SPACE . join  $SPACE, @{
$depth_cutoffs_ref* };

If you give it an array this time it produces "*--cov-threshold 25 100*",
which I think is what you want, right?

I'll go for this and then you can check if it's correct!


Chiara

**Comment by henrikstranneheim** [comment]
Great work! :1st_place_medal:

**Comment by northwestwitch** [comment]
Thanks! :)

I'll do another!

Chiara

**Pull request merged**